api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractService.java

```diff
@@ -60,56 +61,60 @@
      * @param streamer The user to retrieve contracts by.
      * @param state The state to retrieve contracts by.
      * @param pageable identifies the page number and pagesize to retrieve
+     * @param isLoggedIn flag that indicates if the user is logged in.
      * @return a set of all contracts in the given state for the given user.
      */
-    Page<ContractDto> getContractsForStreamerAndState(UserModel streamer, ContractState state, Pageable pageable);
+    Page<Contract> getContractsForStreamerAndState(UserModel streamer, ContractState state, Pageable pageable, boolean isLoggedIn);
```

Comment:
when I read 'isLoggedIn' i wasnt sure if it was for any logged in user or only the loggedIn streamer... Im struggling to think of a better name. Maybe something like 'isPrivate' or 'requestedByStreamer' could be a little more clear.

A different option would be to pass the requester's twitchUsername to the service as a string. Then in the service method check if the requester is the contract's owner, and if it is, return the privateDto.

I favor passing the requester's twitchUsername the most... What do you think?

Reply:
Little bit confused by "any logged in user or only the logged in streamer". They are the same thing right?